src/Lean/Server/FileWorker/RequestHandling.lean

```diff
@@ -117,44 +116,55 @@ partial def handleDefinition (kind : GoToKind) (p : TextDocumentPositionParams)
     if let some modUri ← documentUriFromModule name then
       let range := { start := ⟨0, 0⟩, «end» := ⟨0, 0⟩ : Range }
       let ll : LocationLink := {
-        originSelectionRange? := none
+        originSelectionRange? := (·.toLspRange text) <$> i.stx[2].getRange?
```

Comment:
We still don't make a distinction between positions in the source file and "positions" from the delaborator, so this will give very nonsensical results for delaborated syntax.

```suggestion
        originSelectionRange? := (·.toLspRange text) <$> i.stx[2].getRange? true
```